sunpy/map/mapbase.py

```diff
@@ -1826,6 +1835,13 @@ def draw_rectangle(self, bottom_left, *, width: u.deg=None, height: u.deg=None,
 
         width = Longitude(top_right.spherical.lon - bottom_left.spherical.lon)
         height = Latitude(top_right.spherical.lat - bottom_left.spherical.lat)
+        if width < 0:
```

Comment:
As an alternative to moving this from the utility into here, why not add a flag to the utility which disables this warning. I suggest this mainly because the idea of the utility was so that it could be used in other places (I plan to use it in the `dkist` package where the warning will be useful).

Reply:
I'm not sure that the warning is appropriate for the utility, whose job it is to
- Convert `top_right` into the frame of `bottom_left`
or
- Add `width` and `height` to `bottom_left`.

Whether a warning makes sense then depends what you want to do with the coordinates. In addition, as you say above you can easily construct WCSes where x-y don't correspond to lon-lat, so you really need a WCS for context to see if it makes sense to raise a warning.

Reply:
hmm, yeah ok, it probably is better done in the context of the calling function. Maybe a little write up of this to the `Notes` section of the utility docstring so we don't wonder why it's not there in 3 months time :grinning: ?

Reply:
> you can easily construct WCSes where x-y don't correspond to lon-lat

I'll point out that this utility as written converts `bottom_left` to spherical coordinates and adds width as longitude and height as latitude (even if "longitude" and "latitude" aren't the correct terms in that coordinate frame).  Whether the WCS is something other than spherical (e.g., Cartesian) is ignored.